src/Web/Hyperbole/Application.hs

```diff
@@ -47,24 +48,25 @@ import Web.View (View, renderLazyByteString, renderUrl)
 >   liveApp (basicDocument "Example") $ do
 >      page mainPage
 -}
-liveApp :: (BL.ByteString -> BL.ByteString) -> Eff '[Hyperbole, Server, IOE] Response -> Wai.Application
+liveApp :: (BL.ByteString -> BL.ByteString) -> Eff '[Hyperbole, Server, Concurrent, IOE] Response -> Wai.Application
 liveApp toDoc app =
   websocketsOr
     defaultConnectionOptions
-    (runEff . socketApp app)
+    (runEff . runConcurrent . socketApp app)
     (waiApp toDoc app)
 
 
-socketApp :: (IOE :> es) => Eff (Hyperbole : Server : es) Response -> PendingConnection -> Eff es ()
+socketApp :: (IOE :> es, Concurrent :> es) => Eff (Hyperbole : Server : es) Response -> PendingConnection -> Eff es ()
 socketApp actions pend = do
   conn <- liftIO $ WS.acceptRequest pend
   forever $ do
-    runServerSockets conn $ runHyperbole actions
+    msg <- liftIO $ WS.receiveData conn
+    void $ forkIO $ void $ runServerSockets conn msg $ runHyperbole actions
```

Comment:
I think we might be able to shoot ourselves in the foot with `forkIO` here. My read of the docs says it won't propogate exceptions, among other things. Let's read up on https://hackage.haskell.org/package/effectful-2.3.1.0/docs/Effectful-Concurrent-Async.html

Reply:
Absolutely, I added `forkIO` as a quick fix to see how this will affect and had this on my todo list. Async (effectful, or outside of effects machinery) sounds like a good solution. The question is how to react to exceptions, i.e. if this should kill the socket, etc.